packages/flutter_tools/templates/app/android-java.tmpl/app/src/main/java/androidIdentifier/MainActivity.java.tmpl

```diff
@@ -1,21 +1,9 @@
 package {{androidIdentifier}};
 
 {{#useAndroidEmbeddingV2}}
-{{#androidX}}
-import androidx.annotation.NonNull;
-{{/androidX}}
-{{^androidX}}
-import android.support.annotation.NonNull;
-{{/androidX}}
 import io.flutter.embedding.android.FlutterActivity;
-import io.flutter.embedding.engine.FlutterEngine;
-import io.flutter.plugins.GeneratedPluginRegistrant;
 
 public class MainActivity extends FlutterActivity {
```

Comment:
If you want bonus points, we can actually toggle the AndroidManifest to straight up point the activity to io.flutter.embedding.android.FlutterActivity :) but that probably takes a few more tooling changes

Reply:
I brought this up with I think @blasten earlier. He mentioned that it would be relatively difficult to change to tool to just never generate a file like this based on the embedding flag. I think it's okay to generate an empty MainActivity, it provides an easy entry point for anyone who wants to customize it.

Reply:
ah that was the piece I was missing, whether the tool makes it hard to toggle. This LG then.